### ISSUE TITLE
`missing_panics_doc` falsely flags `unwrap()` and similar fns inside `const` blocks

### ISSUE THREAD

**Issue by itsjunetime** [rust-lang/rust-clippy#15169]
### Summary

The `missing_panics_doc` lint falsely flags instances of `.unwrap()`, `.expect()`, etc, inside `const { }` blocks even though those are guaranteed to be evaluated before runtime (and thus cannot actually cause a panic).

### Lint Name

missing_panics_doc

### Reproducer

I tried this code:

```rust
/// Docs here
pub fn should_pass() {
    let a = const { std::num::NonZeroUsize::new(1).unwrap() };
}
```

I saw this happen:

```
error: docs for function which may panic missing `# Panics` section
 --> src/main.rs:4:1
  |
4 | pub fn should_pass() {
  | ^^^^^^^^^^^^^^^^^^^^
  |
note: first possible panic found here
 --> src/main.rs:5:21
  |
5 |     let a = const { core::num::NonZeroU32::new(1).unwrap() };
  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_panics_doc
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![deny(clippy::missing_panics_doc)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I expected to see this happen:
Clippy should produce no errors or warnings.

### Version

```text
rustc 1.90.0-nightly (bdaba05a9 2025-06-27)
binary: rustc
commit-hash: bdaba05a953eb5abeba0011cdda2560d157aed2e
commit-date: 2025-06-27
host: aarch64-unknown-linux-gnu
release: 1.90.0-nightly
LLVM version: 20.1.7
```
